lib/embulk/guess/avro.rb

```diff
@@ -1,61 +1,57 @@
 module Embulk
   module Guess
+    class Avro < GuessPlugin
+      Plugin.register_guess("avro", self)
 
-    # TODO implement guess plugin to make this command work:
-    #      $ embulk guess -g "avro" partial-config.yml
-    #
-    #      Depending on the file format the plugin uses, you can use choose
-    #      one of binary guess (GuessPlugin), text guess (TextGuessPlugin),
-    #      or line guess (LineGuessPlugin).
+      def guess_columns(sample_buffer)
+        classpath = File.expand_path('../../../../classpath', __FILE__)
+        Dir["#{classpath}/*.jar"].each {|jar| require jar }
+        java_import org.apache.avro.file.DataFileReader
+        java_import org.apache.avro.file.SeekableByteArrayInput
+        java_import org.apache.avro.generic.GenericDatumReader
+        java_import org.embulk.spi.type.Types
 
-    #class Avro < GuessPlugin
-    #  Plugin.register_guess("avro", self)
-    #
-    #  def guess(config, sample_buffer)
-    #    if sample_buffer[0,2] == GZIP_HEADER
-    #      guessed = {}
-    #      guessed["type"] = "avro"
-    #      guessed["property1"] = "guessed-value"
-    #      return {"parser" => guessed}
-    #    else
-    #      return {}
-    #    end
-    #  end
-    #end
+        def convert_type(field)
+          type_map = {
```

Comment:
This value is invariant and it doesn't depend on other variables.
I think that it should be defined as a constant of this class.